utils.py

```diff
@@ -0,0 +1,283 @@
+                )
+                continue
+            column_data_types, dependent_models = self.get_model_datatypes(model_class)
+            meta_tree[f'{app_model.app_name}.{app_model.model_name}'] = {
+                'dependency': sorted(dependent_models),
+                'required_by': [],
+                'metadata': column_data_types,
+                'model': model_class,
+                'app_model': app_model,
+            }
+
+        dependency_map = {
+            model_name: set(model_data["dependency"])
+            for model_name, model_data in meta_tree.items()
+        }
+        for model_name, model_data in meta_tree.items():
+            for dependency in model_data["dependency"]:
+                if dependency in meta_tree:
+                    meta_tree[dependency]["required_by"].append(model_name)
+
+        models_to_export = self.topological_sort(dependency_map, all_models)
+        model_qs_mapper = {}
+        files_to_upload = []
+        count, total = 0, len(models_to_export)
+        self.git_repo.git_checkout()
+        for model in models_to_export:
+            count += 1
+            current_model_app = f'{model.app_name}.{model.model_name}'
+            sys.stdout.write(f'({count}/{total}) {current_model_app}...')
+            json_data = meta_tree[current_model_app]
+            model_class = json_data.pop('model')
+            app_model = json_data.pop('app_model')
+            queryset = self.prepare_queryset(model_class, app_model)
+            if json_data['dependency'] and model.inherit_parent_filter:
+                # This model depends on other models, get its parent queryset
+                inherit_filters = json.loads(model.inherit_parent_filter)
+                parent_qs = [
+                    (model_qs_mapper[dep], condition)
+                    for dep, condition in inherit_filters.items()
+                    if dep in model_qs_mapper and dep != current_model_app
+                ]
+                queryset = self.filter_related_data(queryset, parent_qs)
+
+            # Store qs if model has any dependent model
+            if json_data['required_by']:
+                model_qs_mapper[current_model_app] = queryset
+
+            json_data['data'] = self.serialize_data(queryset, model)
+
+            folder = f'{self.local_data_dir}/{model.report_name}'
+            self.save_json(json_data, folder, current_model_app)
+            files_to_upload.append(f'{model.report_name}/{current_model_app}.json')
+            sys.stdout.write('Done\n')
+
+        if push_option in ('all', 'bucket'):
+            sys.stdout.write('Pushing to Object Store...\n')
+            for file in files_to_upload:
+                self.object_storage.upload_files_to_object_storage(
+                    file, self.bucket_folder_location
+                )
+
+        if push_option in ('all', 'github'):
+            sys.stdout.write('Pushing to Git...\n')
+            self.git_repo.git_commit(files_to_upload)
```

Comment:
The method def process should look like psuedo code (minimal logic and, wrap complex logic in a function)